golem-worker-service-base/src/parser/place_holder_parser.rs

```diff
@@ -11,13 +11,69 @@ pub struct PlaceHolder(pub String);
 impl GolemParser<PlaceHolder> for PlaceHolderPatternParser {
     fn parse(&self, str: &str) -> Result<PlaceHolder, ParseError> {
         match parse_place_holder(str) {
-            Ok(value) => Ok(value.1),
+            Ok((_, parsed)) => Ok(PlaceHolder(parsed.to_string())),
             Err(err) => Result::Err(ParseError::Message(err.to_string())),
         }
     }
 }
 
-pub fn parse_place_holder(input: &str) -> IResult<&str, PlaceHolder> {
-    delimited(tag("{"), take_while(|c| c != '}'), tag("}"))(input)
-        .map(|(rest, captured)| (rest, PlaceHolder(captured.to_string())))
+pub fn parse_place_holder(input: &str) -> IResult<&str, &str> {
+    delimited(tag("{"), take_until_unbalanced('{', '}'), tag("}"))(input)
```

Comment:
Ok, got it. To handle cases such as `{ vari{}able }` ?

Reply:
This is a single pass parser, and is extremely fast. It makes sure we stop parsing on bad input. 

This would prevent parsing bad input like these tests

```rust
    assert!(parse_place_holder("{test").is_err());
    assert!(parse_place_holder("test}").is_err());
    assert!(parse_place_holder("}").is_err());
```

but also stops parsing once we have found all the brackets, which helps it compose with the AllPathPatterns parser